shared/redux/rtk-apis/tasksAPI.ts

```diff
@@ -44,6 +44,10 @@ export const api = createApi({
         method: "DELETE",
       }),
     }),
+
+    getSingleTask: builder.query<TTask, string>({
+      query: (id) => `todos/:${id}`,
```

Comment:
```suggestion
      query: (id) => `todos/${id}`,
```

You don't need the colon. It is used in the backend to indicate that the param taken will be dyanmic